felix/fv/rpf_test.go

```diff
@@ -0,0 +1,161 @@
+			options      infrastructure.TopologyOptions
+			calicoClient client.Interface
+			w            *workload.Workload
+			cc           *Checker
+		)
+
+		BeforeEach(func() {
+			infra = getInfra()
+			options = infrastructure.DefaultTopologyOptions()
+		})
+
+		AfterEach(func() {
+			infra.Stop()
+		})
+
+		JustBeforeEach(func() {
+			felixes, calicoClient = infrastructure.StartNNodeTopology(1, options, infra)
+
+			wIP := "10.65.0.2"
+			w = workload.Run(felixes[0], "w0", "default", wIP, "8055", "udp")
+			w.WorkloadEndpoint.Labels = map[string]string{"name": w.Name}
+			w.ConfigureInInfra(infra)
+
+			pol := api.NewGlobalNetworkPolicy()
+			pol.Namespace = "fv"
+			pol.Name = "policy-1"
+			pol.Spec.Ingress = []api.Rule{{Action: "Allow"}}
+			pol.Spec.Egress = []api.Rule{{Action: "Allow"}}
+			pol.Spec.Selector = "all()"
+
+			pol = func(policy *api.GlobalNetworkPolicy) *api.GlobalNetworkPolicy {
+				log.WithField("policy", dumpResource(policy)).Info("Creating policy")
+				policy, err := calicoClient.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+				Expect(err).NotTo(HaveOccurred())
+				return policy
+			}(pol)
+
+			cc = &Checker{
+				CheckSNAT: true,
+				Protocol:  "udp",
+			}
+		})
+
+		JustAfterEach(func() {
+			if CurrentGinkgoTestDescription().Failed {
+				for _, felix := range felixes {
+					felix.Exec("iptables-save", "-c")
+					felix.Exec("ip", "link")
+					felix.Exec("ip", "addr")
+					felix.Exec("ip", "rule")
+					felix.Exec("ip", "route")
+				}
+			}
+		})
+
+		It("should not allow packets from wrong direction with non-strict RPF on main device", func() {
+			By("turning off RPF per device", func() {
+				felixes[0].Exec("sysctl", "-w", "net.ipv4.conf.all.rp_filter=0")
+				felixes[0].Exec("sysctl", "-w", "net.ipv4.conf.default.rp_filter=0")
+				felixes[0].Exec("sysctl", "-w", "net.ipv4.conf.eth0.rp_filter=0")
+			})
+
+			var eth20 *workload.Workload
+
+			By("setting up node's fake external ifaces", func() {
+				// We name the ifaces ethXY since such ifaces are
+				// treated by felix as external to the node
+				//
+				// Using a test-workload creates the namespaces and the
+				// interfaces to emulate the host NICs
+
+				eth20 = &workload.Workload{
+					Name:          "eth20",
+					C:             felixes[0].Container,
+					IP:            "192.168.20.1",
+					Ports:         "57005", // 0xdead
+					Protocol:      "udp",
+					InterfaceName: "eth20",
+				}
+				err := eth20.Start()
+				Expect(err).NotTo(HaveOccurred())
+
+				// assign address to eth20 and add route to the .20 network
+				felixes[0].Exec("ip", "route", "add", "192.168.20.0/24", "dev", "eth20")
+				felixes[0].Exec("ip", "addr", "add", "10.0.0.20/32", "dev", "eth20")
+				_, err = eth20.RunCmd("ip", "route", "add", "10.0.0.20/32", "dev", "eth0")
+				Expect(err).NotTo(HaveOccurred())
+				// Add a route to the test workload to the fake external
+				// client emulated by the test-workload
```

Comment:
Can you reword this to distinguish more clearly between "test workload" (`w` I think) and "test-workload" (`eth20` I think).

Reply:
Sorry, I wasn't clear, but I was hoping for a clarification of the comment wording...